plugins/services/src/js/components/modals/ServiceRootGroupModal/index.tsx

```diff
@@ -196,8 +243,36 @@ class ServiceRootGroupModal extends React.Component<
     });
   }
 
+  getGroupFormData(): void {
+    const { id } = this.props;
+    if (!!id) {
+      getGroup(id)
+        .pipe(take(1))
+        .subscribe({
+          next: groupData => {
+            const data = groupFormDataFromGraphql(groupData.data.group);
+            this.setState({
+              data,
+              originalData: JSON.parse(JSON.stringify(data))
+            });
+          }
+        });
+    }
+    this.setState({
+      data: emptyGroupFormData()
+    });
+  }
+
   getModalContent() {
-    const { data, errors } = this.state;
+    const { id } = this.props;
+    const { errors, data } = this.state;
+    // If id exists, then we must be editing.
+    const isEdit = !!id;
```

Comment:
we should move `isEdit` to state and just load it from there instead of continually checking. and remove it from props if we aren't using that value.